modules/ixBidAdapter.js

```diff
@@ -92,12 +99,68 @@ function bidToImp(bid) {
     imp.ext.sid = `${bid.params.size[0]}x${bid.params.size[1]}`;
   }
 
-  if (bid.params.hasOwnProperty('bidFloor') && bid.params.hasOwnProperty('bidFloorCur')) {
-    imp.bidfloor = bid.params.bidFloor;
-    imp.bidfloorcur = bid.params.bidFloorCur;
+  return imp;
+}
+
+/**
+ * Gets priceFloors floors and IX adapter floors,
+ * Validates and sets the higher one on the impression
+ * @param  {object}    bid bid object
+ * @param  {object}    imp impression object
+ * @param  {string}    mediaType the impression ad type, one of the SUPPORTED_AD_TYPES
+ */
+function _applyFloor(bid, imp, mediaType) {
+  let adapterFloor = null;
+  let moduleFloor = null;
+
+  if (bid.params.bidFloor && bid.params.bidFloorCur) {
+    adapterFloor = { floor: bid.params.bidFloor, currency: bid.params.bidFloorCur };
   }
 
-  return imp;
+  if (utils.isFn(bid.getFloor)) {
+    let _mediaType = '*';
+    let _size = '*';
+
+    if (mediaType && utils.contains(SUPPORTED_AD_TYPES, mediaType)) {
+      const { w: width, h: height } = imp[mediaType];
+      _mediaType = mediaType;
+      _size = [width, height];
+    }
+    moduleFloor = bid.getFloor({
```

Comment:
I recommend surrounding this in a try catch. Just in case the price floors module throws up. We do not want your entire request to get thrown out!

you do not have to of course, just a suggestion.